### PR TITLE
Allow workflow reordering

### LINKED ISSUE #2510: Allow project owner to specify what order the work flow buttons for mutiple workflows show on the landing page
Currently if you have multiple workflows the buttons on the landing page are determined by the order of creation, but there may be a time where one wants to promote one over the other and switch the order. Currently there is no way to do that. The order the workflows show on the landing page should be customizable . Like for Comet Hunters well want to prioritize the Hype Suprime-Cam data, so it would be nice if I can put the HSC workflow button above the current workflow on Comet Hunters when we finally add in our second workflow

screen shot 2016-04-29 at 10 02 16 pm

Order is in creation order 
screen shot 2016-04-29 at 10 07 15 pm

### PR DESCRIPTION
**Pull request by brian-c**
Allows setting workflow order in the project builder and displays workflows in that order on the project home page.

@camallen I'm storing this order on the project configuration. There's no way to store it in the project's workflow links, correct?

Closes #2510.

### CODE REVIEW

**Comment by camallen** [comment]
@brian-c not the workflow has no information about how the project should order them, def belongs on the project object in my mind (as i may want to copy the workflow data to another project).

File: app/lib/get-workflows-in-order.coffee [hunk 1]
```diff
@@ -0,0 +1,23 @@
+getWorkflowsInOrder = (project, query) ->
+  console.log {project}
```
Comment (simensta) [inline app/lib/get-workflows-in-order.coffee]: leftover log on ln2

File: app/pages/lab/project.cjsx [hunk 2]
```diff
@@ -82,24 +93,28 @@ EditProjectPage = React.createClass
...
+                <ChangeListener key={workflow.id} target={workflow} eventName="save" handler={renderWorkflowListItem.bind null, workflow} />
 
-                <li className="nav-list-item">
-                  <button type="button" onClick={@showCreateWorkflow} disabled={@state.workflowCreationInProgress} title="A workflow is the sequence of tasks that you’re asking volunteers to perform.">
-                    New workflow{' '}
-                    <LoadingIndicator off={not @state.workflowCreationInProgress} />
-                  </button>
-                </li>
-              </ul>
+              <DragReorderable tag="ul" className="nav-list" items={workflows} render={renderWorkflowListItemListener} onChange={@handleWorkflowReorder} />
```
Comment (simensta) [inline app/pages/lab/project.cjsx]: Is the onChange handler in ln109 missing the `newOrder` arg?

Comment (brian-c) [inline app/pages/lab/project.cjsx]: I'm not sure I follow. The `onChange` function is called (with its `newOrder`) from within the component's drop handler. This change handler is defined on line 38.

Comment (brian-c) [inline app/pages/lab/project.cjsx]: Ah, I think I got it now. I guess ChangeListeners cannot be the direct children of a DragReorderable.

**Comment by simensta** [comment]
@brian-c: In a workflow tile, are both the `<li>` and the `<a>` intended to be draggable? That isn't consistent for me across browsers. In Chrome, I can reorder by dragging the `<li>` or the `<a>`. In Firefox, attempting reorder by dragging the `<a>` loads the workflow page for that tile, without reordering the workflows.

**Comment by brian-c** [comment]
Rebased. I'm fine with the browser difference for now. I've opened an issue in the drag-reorderable lib.

**Comment by brian-c** [comment]
Just noticed this hasn't been merged, gonna do it myself since @mschwamb needs it for tomorrow (see #2510).

**Comment by mschwamb** [comment]
Thanks @brian-c  I'm trying to see where/what I need to do to reorder. or Maybe it hasn't gotten onto the production version yet. If you don't see the HSC Search as the top button can you help reorder. I added you as a collaborator so it will be easier to find

**Comment by brian-c** [comment]
Sorry @mschwamb, we've been having issues with our automatic deploy process. I've done a manual deploy and set HSC Search as the first workflow. Should be all set now.

**Comment by mschwamb** [comment]
Thanks @brian-c

**Pull request merged**